.github/workflows/tests.yml

```diff
@@ -0,0 +1,97 @@
+name: "Tests"
+
+on:
+  pull_request:
+    branches:
+      - "v*.*"
+      - "master"
+  push:
+    branches:
+      - "v*.*"
+      - "master"
+
+jobs:
+  phpunit:
+    name: "PHPUnit tests"
+    runs-on: "${{ matrix.os }}"
+
+    strategy:
+      fail-fast: true
+      matrix:
+        os:
+          - "ubuntu-20.04"
+        php-version:
+          - "7.4"
+          - "8.0"
+        mongodb-version:
+          - "4.4"
+        driver-version:
+          - "stable"
+        topology:
+          - "server"
+        include:
+          - os: "ubuntu-20.04"
+            php-version: "8.0"
+            mongodb-version: "4.4"
+            driver-version: "stable"
+            topology: "replica_set"
+          - os: "ubuntu-20.04"
+            php-version: "8.0"
+            mongodb-version: "4.4"
+            driver-version: "stable"
+            topology: "sharded_cluster"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v2"
+        with:
+          fetch-depth: 2
+
+      - id: setup-mongodb
+        uses: mongodb-labs/drivers-evergreen-tools@master
+        with:
+          version: ${{ matrix.mongodb-version }}
+          topology: ${{ matrix.topology }}
+
+      - name: Setup cache environment
+        id: extcache
+        uses: shivammathur/cache-extensions@v1
+        with:
+          php-version: ${{ matrix.php-version }}
+          extensions: "mongodb-${{ matrix.driver-version }}"
+          key: "extcache-v1"
+
+      - name: Cache extensions
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.extcache.outputs.dir }}
+          key: ${{ steps.extcache.outputs.key }}
+          restore-keys: ${{ steps.extcache.outputs.key }}
+
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          php-version: "${{ matrix.php-version }}"
+          tools: "pecl"
+          extensions: "mongodb-${{ matrix.driver-version }}"
+          coverage: "none"
+          ini-values: "zend.assertions=1"
+
+      - name: "Show driver information"
+        run: "php --ri mongodb"
+
+      - name: "Cache dependencies installed with composer"
+        uses: "actions/cache@v2"
+        with:
+          path: "~/.composer/cache"
+          key: "php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock') }}"
+          restore-keys: "php-${{ matrix.php-version }}-composer-locked-"
+
+      - name: "Install dependencies with composer"
+        run: "composer update --no-interaction --no-progress"
+
+      - name: "Run PHPUnit"
+        run: "vendor/bin/simple-phpunit -v"
+        env:
+          SYMFONY_DEPRECATIONS_HELPER: 999999
```

Comment:
Would it make more sense to just use "disabled" here? See: https://symfony.com/blog/new-in-symfony-3-1-deprecation-helper-improvements#ignoring-triggered-deprecations

Reply:
`disabled` disables logging of deprecations, while this setting logs deprecations but doesn't set a non-zero exit code for `phpunit` until 1000000 deprecations are reached in a run.

Reply:
Got it. So this gives you a chance to view the deprecations in build logs without actually failing the build. Works for me if you'll find that useful.